libs/agno/agno/agent/agent.py

```diff
@@ -5792,6 +5817,37 @@ def _get_effective_filters(self, knowledge_filters: Optional[Dict[str, Any]] = N
 
         return effective_filters
 
+    def get_messages_from_previous_sessions(self, number_of_sessions: int) -> List[Message]:
+        """Internal helper method to get messages from previous sessions."""
+        if self.storage is None:
+            return []
+
+        sessions = self.storage.get_all_sessions(user_id=self.user_id)
```

Comment:
Lets add a `get_last_n_sessions` which automatically takes the created at into account